tests-e2e/core/argocd_gitopsengine_instance_test.go

```diff
@@ -0,0 +1,140 @@
+	"github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture"
+	"github.com/redhat-appstudio/managed-gitops/tests-e2e/fixture/k8s"
+	apps "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var _ = Describe("ArgoCD instance via GitOpsEngineInstance Operations Test", func() {
+
+	const (
+		argocdNamespace = fixture.NewArgoCDInstanceNamespace
+	)
+
+	Context("ArgoCD instance gets created from an operation's gitopsEngineInstance resource-type", func() {
+
+		BeforeEach(func() {
+			By("Delete old namespaces, and kube-system resources")
+			Expect(fixture.EnsureCleanSlate()).To(Succeed())
+
+		})
+
+		It("ensures that a standalone ArgoCD gets created successfully when an operation CR of resource-type GitOpsEngineInstance is created", func() {
+			err := unsafeCleanDBWhereIDisAutoGenerated()
+			Expect(err).To(BeNil())
+			if fixture.IsRunningAgainstKCP() {
+				Skip("Skipping this test until we support running gitops operator with KCP")
+			}
+
+			dbq, err := db.NewUnsafePostgresDBQueries(true, true)
+			Expect(err).To(BeNil())
+			defer dbq.CloseDatabase()
+			err = db.SetupForTestingDBGinkgo()
+			Expect(err).To(BeNil())
+
+			k8sClient, err := fixture.GetE2ETestUserWorkspaceKubeClient()
+			Expect(err).To(Succeed())
+
+			testClusterUser := &db.ClusterUser{
+				Clusteruser_id: "test-user",
+				User_name:      "test-user",
+			}
+			ctx := context.Background()
+			log := log.FromContext(ctx)
+
+			By("Creating gitopsengine cluster,cluster user and namespace")
+
+			newArgoCDNamespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: argocdNamespace,
+				},
+			}
+			err = k8sClient.Create(ctx, newArgoCDNamespace)
+			Expect(err).To(BeNil())
+
+			err = util.CreateNewArgoCDInstance(ctx, newArgoCDNamespace, *testClusterUser, k8sClient, log, dbq)
+			Expect(err).To(BeNil())
+
+			By("ensuring ArgoCD service resource exists")
+			argocdInstance := &apps.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: newArgoCDNamespace.Name + "-server", Namespace: newArgoCDNamespace.Name},
+			}
+
+			Eventually(argocdInstance, "10m", "5s").Should(k8s.ExistByName(k8sClient), "Argo CD server Deployment should exist")
+			err = unsafeCleanDBWhereIDisAutoGenerated()
+			Expect(err).To(BeNil())
+
+		})
+	})
+})
+
+func unsafeCleanDBWhereIDisAutoGenerated() error {
+	ctx := context.Background()
+
+	// 'testSetup' deletes all database rows that might interfere with ArgoCD setup into a namespace
+	// that the OperationCR of resource_type GitopsEngineInstance point to.
+	// This ensures a clean slate for the test run.
+
+	dbq, err := db.NewUnsafePostgresDBQueries(true, true)
+	Expect(err).To(BeNil())
+
+	defer dbq.CloseDatabase()
+	var clusterAccess []db.ClusterAccess
+	err = dbq.UnsafeListAllClusterAccess(ctx, &clusterAccess)
+	Expect(err).To(BeNil())
+
+	for _, clusterAccess := range clusterAccess {
+		if clusterAccess.SeqID >= 0 {
```

Comment:
@raghavi101 Can you explain the significance of checking if `SeqID >= 0`? 

I would have thought that ALL the entries in the database table would have a SeqID >= 0 (since this value should be autogenerated by postgresql in all cases, and since the value starts at 0), which means this code will just delete all the ClusterAccesses, and all the Operations (rather than only those that are specific to the E2E tests).

Reply:
@jgwest This would be hazardous but I could not find a way other than this 
- which will clean the gitopsEngineInstance values. As we are using GetOrCreate for gitopsEngineInstance, if the code GETS a gitopsEngineInstance it will pick that up, namespace will be as present in its db entry overall discarding the namespace passed in the e2e and the test would fail
- after running the test there will be only one gitopsEngineInstance entry left and other tests will get affected because the altered gitopsEngineInstance will be picked and the db entries will be mismatched as required by those tests